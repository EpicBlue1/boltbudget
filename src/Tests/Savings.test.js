import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SavingsSec from '../components/Savings';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

describe('Testing Overview functions excluding savings section', () => {

    const MockUsersTotal = 56000;
    const MockUsersTotAfterTex = 48000;

    const userInput = ({inputOne, inputTwo}) => {

        const checkElement = screen.getByLabelText(/CheckBox/i);
        const inputPercElement = screen.getByTestId(/PercInput/i);
        
        if(inputOne){
            userEvent.type(inputPercElement, inputOne);
        }
        if(inputTwo){
            userEvent.type(inputPercElement, inputTwo);
        }

        return {
            inputPercElement,
            checkElement
        }
    }

    beforeEach(() => {
        console.log("Savings Functions Test")
        render(<SavingsSec totalIncome={MockUsersTotal} totAfterTax={MockUsersTotAfterTex}/>);
    })

    test('Make sure input is empty and checkbox is default', () => {
        const {inputPercElement, checkElement} = userInput({});

        userInput({})

        expect(inputPercElement.value).toBe("");
    })

    test('Check box Before checked and if value changes to after tax', () => {
        const {checkElement} = userInput({});

        const checkBox = screen.getByLabelText(/CheckBox/i);
        const Value = screen.getByTestId(/TotalToSave/i);

        expect(Value.textContent).toBe("R 56000");
    })

    test("Test calculation when checkbox checked", () =>{
    
        const {inputPercElement} = userInput({inputOne: "10"});

        const percCalBut = screen.getByTestId(/CalculatePerc/i);


        userEvent.click(percCalBut);

        const resultElement = screen.getByTestId(/Saved/i);

        expect(resultElement.textContent).toBe("R 5600")
    })

    test('Test if check box is checked and if value changes to Before Tax', () => {
        const {checkElement} = userInput({});

        const checkBox = screen.getByLabelText(/CheckBox/i);
        const Value = screen.getByTestId(/TotalToSave/i);

        userEvent.click(checkBox);

        expect(Value.textContent).toBe("R 8000");
    })

    test("Test calculation when checkbox checked", () =>{
    
        const {inputPercElement} = userInput({inputOne: "10"});

        const checkBox = screen.getByLabelText(/CheckBox/i);
        const percCalBut = screen.getByTestId(/CalculatePerc/i);

        userEvent.click(checkBox);

        userEvent.click(percCalBut);

        const resultElement = screen.getByTestId(/Saved/i);

        expect(resultElement.textContent).toBe("R 800")
    })

    it('If Savings renders correctly', () => {
        const component = renderer.create(<SavingsSec totalIncome={MockUsersTotal} totAfterTax={MockUsersTotAfterTex}/>).toJSON();
    
        expect(component).toMatchSnapshot();
    });
    
});