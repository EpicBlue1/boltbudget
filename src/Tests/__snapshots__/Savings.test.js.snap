// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Overview functions excluding savings section If Savings renders correctly 1`] = `
<div
  className="SavSec"
  id="section3"
>
  <div
    className="SavingsCol float"
  >
    <p
      className="float"
    >
      Calculate after tax
    </p>
    <input
      aria-label="CheckBox"
      className="CHECK"
      onClick={[Function]}
      type="checkbox"
    />
    <br />
    <br />
    <h2
      className="SavingHeading"
    >
      Percentage
    </h2>
    <input
      data-testid="PercInput"
      id="PercVal"
      placeholder="%"
      type="numer"
    />
    <button
      data-testid="CalculatePerc"
      onClick={[Function]}
    >
      Calculate
    </button>
  </div>
  <div
    className="SavingsCol float"
  >
    <h2
      className="SavingHeading"
    >
      Potential to save
    </h2>
    <p
      data-testid="TotalToSave"
    >
      R 
      56000
    </p>
    <h2
      className="SavingHeading"
    >
      Total Saved monthly
    </h2>
    <p
      data-testid="Saved"
    >
      R 
      Type in a percentage
    </p>
  </div>
</div>
`;

exports[`Testing Overview functions excluding savings section renders correctly 1`] = `
<div
  className="SavSec"
  id="section3"
>
  <div
    className="SavingsCol float"
  >
    <p
      className="float"
    >
      Calculate after tax
    </p>
    <input
      aria-label="CheckBox"
      className="CHECK"
      onClick={[Function]}
      type="checkbox"
    />
    <br />
    <br />
    <h2
      className="SavingHeading"
    >
      Percentage
    </h2>
    <input
      data-testid="PercInput"
      id="PercVal"
      placeholder="%"
      type="numer"
    />
    <button
      data-testid="CalculatePerc"
      onClick={[Function]}
    >
      Calculate
    </button>
  </div>
  <div
    className="SavingsCol float"
  >
    <h2
      className="SavingHeading"
    >
      Potential to save
    </h2>
    <p
      data-testid="TotalToSave"
    >
      R 
      56000
    </p>
    <h2
      className="SavingHeading"
    >
      Total Saved monthly
    </h2>
    <p
      data-testid="Saved"
    >
      R 
      Type in a percentage
    </p>
  </div>
</div>
`;
